### ISSUE TITLE
BibTeX Key Generator: not possible to change key pattern

### ISSUE THREAD

**Issue by jorgman1** [JabRef/jabref#5825]
JabRef version 5.0-beta.21 on  Windows 7 x64 and Linux Mint 18.1 (Ubuntu 16.04)

- [X] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists (beta.336)

It is not possible to change the default key pattern. When clicking on it, it is possible to change it, but once the focus of the field is lost, it resets to the old value.

**Comment by calixtus** [comment]
Does it fail in library properties or in the global preferences?

**Comment by jorgman1** [comment]
In the global preferences.

I didn't know that one could set it as well in the libary preferences. But it is a good temporary solution. Thanks!

**Comment by calixtus** [comment]
Thanks. I'm going to take a look.

**Comment by calixtus** [comment]
Mystery solved: In order to save the changes, you have to press return in the text field after entering the change. Maybe I'm going to implement an focus lost event here, so it gets saved also if you just click somewhere outside the field. But thanks for the tip anyway.

**Comment by jorgman1** [comment]
Yeah, maybe it is useful to have a focus lost event there saving the last status.

Thank you for your quick response!

**Comment by jorgman1** [comment]
Something similar happens when editing journal abbreviations. One has to press enter to save the changes in that field.

**Comment by Siedlerchr** [comment]
Yes this is weird, but it's actually the default standard for tables with the JavaFX table.
I hate it as well.

**Comment by calixtus** [comment]
Looks like others had this problem too. At least there are already some hacks around.
https://stackoverflow.com/questions/29576577/tableview-doesnt-commit-values-on-focus-lost-event

**Comment by jorgman1** [comment]
You're right. Apparently there is an eight year old [issue](https://bugs.openjdk.java.net/browse/JDK-8089514) in JDK. It doesn't seem that they are going to change this behaviour soon.

Are there tables in Jabref that need to remain unchanged when the focus is lost?

**Comment by wujastyk** [comment]
JabRef 5.0-beta.491--2020-02-26--0839875
Linux 5.3.0-40-generic amd64 
Java 13.0.2

I just had this problem today too and was glad to find the solution here.  I suggest documenting it LOUDLY in the help text: [https://docs.jabref.org/setup/bibtexkeypatterns](https://docs.jabref.org/setup/bibtexkeypatterns)

**Comment by Siedlerchr** [comment]
Feel free to edit the docs, just click on edit at github and you can edit the content

**Comment by mikegilchrist** [comment]
Even if I hit enter, changes to my BibTeX Key Patterns aren't saved.  

Here's my install information.  I downloaded the latest .deb version from FOSSHub today.

JabRef 5.0--2020-03-06--2e6f433
Linux 4.15.0-101-generic amd64 
Java 13.0.2

**Comment by AEgit** [comment]
@mikegilchrist Could you try the latest dev version (the one you used is already a bit out of date)? The latest dev version can be downloaded here:
https://builds.jabref.org/master/

Always remember to make a backup of your JabRef library before using developer versions.

**Comment by mikegilchrist** [comment]
@AEgit  Yes, it appears that the failure to update the preference entry and apply it until you restart the program are fixed in the dev version 

JabRef 5.1--2020-05-19--fe5eec6
Linux 4.15.0-101-generic amd64 
Java 14.0.1

Thanks.

**Comment by AEgit** [comment]
Cheers.

**Comment by Siedlerchr** [comment]
Idea for a workaround: Add a hint in the table
https://github.com/JabRef/jabref/issues/6770#issuecomment-676480063

**Comment by calixtus** [comment]
Hint: ControlsFX provides a TableView2 as a drop-in replacement for TableView claiming having solved this issue. However, this issue has had a longer discussion in the jdk bugtracker.
See this discussion on stackoverflow:
https://stackoverflow.com/questions/24694616/how-to-enable-commit-on-focuslost-for-tableview-treetableview#25291137

**Comment by Siedlerchr** [comment]
From what I see it should be sufficient to use the TextField2TableCell implementation to achieve this

https://github.com/controlsfx/controlsfx/blob/jfx-13/controlsfx/src/main/java/org/controlsfx/control/tableview2/cell/TextField2TableCell.java

**Comment by koppor** [comment]
Link to the JDK bug tracker: https://bugs.openjdk.java.net/browse/JDK-8089514: "[TableView, TreeView, ListView, TreeTableView] Clicking outside of the edited cell, node, or entry should commit the value"
